code/modules/mod/mod_modules.dm

```diff
@@ -0,0 +1,1441 @@
+	use_power_cost = 50
+	incompatible_modules = list(/obj/item/mod/module/constructor)
+	cooldown_time = 11 SECONDS
+
+/obj/item/mod/module/constructor/on_equip()
+	ADD_TRAIT(mod.wearer, TRAIT_QUICK_BUILD, MOD_TRAIT)
+
+/obj/item/mod/module/constructor/on_unequip()
+	REMOVE_TRAIT(mod.wearer, TRAIT_QUICK_BUILD, MOD_TRAIT)
+
+/obj/item/mod/module/constructor/on_use()
+	. = ..()
+	if(!.)
+		return
+	rcd_scan(src, fade_time = 10 SECONDS)
+
+/obj/item/mod/module/longfall
+	name = "MOD longfall module"
+	desc = "A module that stops fall damage from happening to the user, converting into kinetic charge."
+	complexity = 1
+	use_power_cost = 100
+	incompatible_modules = list(/obj/item/mod/module/longfall)
+
+/obj/item/mod/module/longfall/on_equip()
+	RegisterSignal(mod.wearer, COMSIG_LIVING_Z_IMPACT, .proc/z_impact_react)
+
+/obj/item/mod/module/longfall/on_unequip()
+	UnregisterSignal(mod.wearer, COMSIG_LIVING_Z_IMPACT)
+
+/obj/item/mod/module/longfall/proc/z_impact_react(datum/source, levels, turf/fell_on)
+	if(!drain_power(use_power_cost*levels))
+		return
+	new /obj/effect/temp_visual/mook_dust(get_turf(mod.wearer))
+	mod.wearer.Stun(levels * 1 SECONDS)
+	to_chat(mod.wearer, span_notice("[src] protects you from the damage!"))
+	return COMPONENT_NO_Z_DAMAGE
+
+/obj/item/mod/module/thermal_regulator
+	name = "MOD thermal regulator module"
+	desc = "A module that lets you set a temperature to keep your body at."
+	icon_state = "regulator"
+	module_type = MODULE_TOGGLE
+	complexity = 2
+	active_power_cost = 5
+	incompatible_modules = list(/obj/item/mod/module/thermal_regulator)
+	cooldown_time = 0.5 SECONDS
+	var/temperature_setting = BODYTEMP_NORMAL
+	var/min_temp = 293.15
+	var/max_temp = 318.15
+
+/obj/item/mod/module/thermal_regulator/get_configuration()
+	. = ..()
+	.["temperature_setting"] = add_ui_configuration("Temperature", "number", temperature_setting - T0C)
+
+/obj/item/mod/module/thermal_regulator/configure_edit(key, value)
+	switch(key)
+		if("temperature_setting")
+			temperature_setting = clamp(value + T0C, min_temp, max_temp)
+
+/obj/item/mod/module/thermal_regulator/on_process(delta_time)
+	. = ..()
+	if(!.)
+		return
+	mod.wearer.adjust_bodytemperature(get_temp_change_amount((temperature_setting - mod.wearer.bodytemperature), 0.08 * delta_time))
+
+/obj/item/mod/module/injector
+	name = "MOD injector module"
+	desc = "A module that extends a piercing injector."
+	module_type = MODULE_ACTIVE
+	complexity = 1
+	use_power_cost = 25
+	device = /obj/item/reagent_containers/syringe/mod
+	incompatible_modules = list(/obj/item/mod/module/injector)
+	cooldown_time = 0.5 SECONDS
+
+/obj/item/reagent_containers/syringe/mod
+	name = "MOD injector syringe"
+	desc = "A piercing injector fitting in a MODsuit."
+	icon_state = "mod_0"
+	base_icon_state = "mod"
+	amount_per_transfer_from_this = 30
+	possible_transfer_amounts = list(5, 10, 15, 20, 30)
+	volume = 30
+	inject_flags = INJECT_CHECK_PENETRATE_THICK
+
+/obj/item/mod/module/circuit
```

Comment:
Convention suggests that this should be moved to its own file. Probably `code/modules/wiremod/shell/mod.dm`

Reply:
yah

Reply:
actually, this is an issue of 2 systems in modular folders colliding, i think i will leave it in the mod folder, just how stuff like bluespace launchpad circuit is handled by the bluespace launchpad file